app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,237 @@
+
+                taskService.createTask(task1);
+                taskService.createTask(task2);
+
+                expectedSize = taskService.getTasks().size();
+            }
+
+            @Test
+            @DisplayName("등록된 모든 할 일을 리턴한다")
+            void it_returns_all_tasks() {
+                // when
+                List<Task> findTasks = subject();
+
+                final int actualSize = findTasks.size();
+
+                // then
+                assertThat(actualSize).isEqualTo(expectedSize);
+                assertThat(findTasks.get(0).getTitle()).isEqualTo(task1.getTitle());
+                assertThat(findTasks.get(1).getTitle()).isEqualTo(task2.getTitle());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("createTask 메소드는")
+    class Describe_createTask {
+        @Nested
+        @DisplayName("새로운 할 일을 추가한다면")
+        class Context_with_new_task extends NewTask {
+            private final Long TASK_ID = 1L;
+            private final String TITLE = "할 일 추가";
+            private final Task givenTask = withTitle(TITLE);
+            private int expectedSize;
+
+            @BeforeEach
+            void prepareTest() {
+                expectedSize = taskService.getTasks().size() + 1;
+            }
+
+            @Test
+            @DisplayName("DB에 등록 후 등록된 할 일을 리턴한다")
+            void it_returns_registered_task() {
+                // when
+                Task actualTask = taskService.createTask(givenTask);
+
+                final int actualSize = taskService.getTasks().size();
+                final Task actualFindTask = taskService.getTask(TASK_ID);
+
+                // then
+                assertThat(actualSize).isEqualTo(expectedSize);
+                assertThat(actualTask.getTitle()).isEqualTo(givenTask.getTitle());
+                assertThat(actualFindTask.getTitle()).isEqualTo(givenTask.getTitle());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("getTask 메소드는")
+    class Describe_getTask {
+        @Nested
+        @DisplayName("존재하는 할 일 ID로 검색한다면")
+        class Context_with_existing_task_id extends NewTask {
+            private final Long TASK_ID = 1L;
+            private final String TITLE = "할 일 검색";
+            private Task givenTask;
+
+            @BeforeEach
+            void prepareTests() {
+                givenTask = withTitle(TITLE);
+                taskService.createTask(givenTask);
+            }
+
+            @Test
+            @DisplayName("검색된 할 일을 리턴한다")
+            void it_returns_found_task() {
+                // when
+                Task actualTask = taskService.getTask(TASK_ID);
+
+                // then
+                assertThat(actualTask.getTitle()).isEqualTo(givenTask.getTitle());
+            }
+
+        }
+
+        @Nested
+        @DisplayName("존재하지않는 할 일 ID로 검색한다면")
+        class Context_with_non_existing_task_id {
+            @Test
+            @DisplayName("TaskNotFoundException 예외가 발생한다")
```

Comment:
다음 두 문장을 비교해 보세요. 어떤 점이 다른가요?

- getTask 메소드는, 존재하지않는 할 일 ID로 검색한다면, TaskNotFoundException 예외가 발생한다.
- getTask 메소드는, 존재하지않는 할 일 ID가 주어지면, 할 일을 찾을 수 없다는 예외를 던진다.

```suggestion
            @DisplayName("TaskNotFoundException 예외를 던진다")
```